### PR TITLE
mermerd: init at 0.6.1.

### PR DESCRIPTION
**Pull request by austin-artificial**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/database/mermerd/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}:
+
+buildGoModule rec {
+  pname = "mermerd";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "KarnerTh";
+    repo = "mermerd";
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/development/tools/database/mermerd/default.nix:13]: The automated update process works smoother if we try not to get in trouble with branch/tags confusion in upstream repos.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/tools/database/mermerd/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "sha256-mQIHv3KjUCtXhzbqBNuooCH7b/LXVUHTZO5n0CJym/A=";
```
Comment (fabaff) [inline pkgs/development/tools/database/mermerd/default.nix:14]: SRI hashes are preferred nowadays.

```suggestion
    hash = "sha256-mQIHv3KjUCtXhzbqBNuooCH7b/LXVUHTZO5n0CJym/A=";
```

File: pkgs/development/tools/database/mermerd/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  vendorHash = "sha256-h1jSPJYFJKkP8kNfD8bn8cmaqVZ20K+S6czau8DnauQ=";
+
+  # the tests expect a database to be running
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Create Mermaid-Js ERD diagrams from existing tables";
+    homepage = "https://github.com/KarnerTh/mermerd";
```
Comment (fabaff) [inline pkgs/development/tools/database/mermerd/default.nix:24]: Tools (e.g., `nix-update`) could add a link to the changelog in the commit messages and/or the PR description if `changelog = ....` is part of meta.

```suggestion
    homepage = "https://github.com/KarnerTh/mermerd";
    changelog = "https://github.com/KarnerTh/mermerd/releases/tag/v${version}";
```

**Comment by flokli** [comment]
@austin-artificial can you rebase this?

**Comment by austin-artificial** [comment]
@flokli Done

**Pull request merged**